x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/persistence/DataFrameAnalyticsConfigProvider.java

```diff
@@ -68,28 +77,116 @@ public DataFrameAnalyticsConfigProvider(Client client, NamedXContentRegistry xCo
         this.xContentRegistry = xContentRegistry;
     }
 
-    public void put(DataFrameAnalyticsConfig config, Map<String, String> headers, ActionListener<IndexResponse> listener) {
-        String id = config.getId();
-
+    /**
+     * Puts the given {@link DataFrameAnalyticsConfig} document into the config index.
+     */
+    public void put(DataFrameAnalyticsConfig config, Map<String, String> headers, ActionListener<DataFrameAnalyticsConfig> listener) {
         if (headers.isEmpty() == false) {
             // Filter any values in headers that aren't security fields
             config = new DataFrameAnalyticsConfig.Builder(config)
                 .setHeaders(filterSecurityHeaders(headers))
                 .build();
         }
+        index(config, null, listener);
+    }
+
+    /**
+     * Updates the {@link DataFrameAnalyticsConfig} document in the config index using given {@link DataFrameAnalyticsConfigUpdate}.
+     */
+    public void update(DataFrameAnalyticsConfigUpdate update,
+                       Map<String, String> headers,
+                       ClusterState clusterState,
+                       ActionListener<DataFrameAnalyticsConfig> listener) {
+        String id = update.getId();
+        GetRequest getRequest = new GetRequest(MlConfigIndex.indexName(), DataFrameAnalyticsConfig.documentId(id));
```

Comment:
We should be able to use the local `get` method to avoid having to handle parsing here.

Reply:
What do you think would be the right place to store `getSeqNo` and `getPrimaryTerm` then?
Should I extend `GetDataFrameAnalyticsAction.Response` with these 2 fields?
Or maybe even extend the base response class (`AbstractGetResourcesResponse`)?

Reply:
Oh, I see. This is why you had to get the doc directly. Sorry, I missed that. Hm, I'd say let's leave this as is then for now.